### PR TITLE
add choice between model_type pbr and synth+real

### PR DESCRIPTION
**Pull request by MedericFourmy**

### CODE REVIEW

File: happypose_ros/test/test_initialization_params.py [hunk 1]
```diff
@@ -103,6 +106,7 @@ def test_leading_publish_tf(happypose_params: dict) -> None:
             **happypose_params,
             parameter_overrides=[
                 Parameter("cosypose.dataset_name", Parameter.Type.STRING, "ycbv"),
+                Parameter("cosypose.model_type", Parameter.Type.STRING, "synth+real"),
```
Comment (Kotochleb) [inline happypose_ros/test/test_initialization_params.py:109]: This test is not required in this case, as we assume the default behaviour is `pbr`. This test only checks fields that are absolutely obligatory for the end user to provide

File: happypose_ros/happypose_ros/happypose_ros_parameters.yaml [hunk 2]
```diff
@@ -44,10 +44,17 @@ happypose_ros:
...
       default_value: ""
-      description: "Name of a dataset used during training."
+      description: "Name of BOP dataset, used to load specific weights and object models."
       validation:
         one_of<>: [["hope", "tless", "ycbv"]]
       read_only: true
+    model_type:
+      type: string
+      default_value: "pbr"
+      description: "What type of NN model to use, type of training data used,  'pbr'|'synth+real' "
```
Comment (Kotochleb) [inline happypose_ros/happypose_ros/happypose_ros_parameters.yaml:54]:
````Suggestion
description: "Type of neural network model to use. Available: 'pbr'|'synth+real'"
```

File: happypose_examples/config/cosypose_params_multiview.yaml [hunk 3]
```diff
@@ -15,16 +15,16 @@
     # Specifies which pose estimator to use in the pipeline
     pose_estimator_type: "cosypose"
     cosypose:
-      # Name of a dataset used during training
+      # Name of BOP dataset, used to load specific weights and object models
       dataset_name: $(var dataset_name)
+      # What type of NN model to use, type of training data used,  'pbr'|'synth+real'
```
Comment (Kotochleb) [inline happypose_examples/config/cosypose_params_multiview.yaml]:
````Suggestion
      # Type of neural network model to use. Available 'pbr'|'synth+real' 
```

File: happypose_examples/config/cosypose_params.yaml [hunk 4]
```diff
@@ -9,16 +9,16 @@
     # Specifies which pose estimator to use in the pipeline
     pose_estimator_type: "cosypose"
     cosypose:
-      # Name of a dataset used during training
+      # Name of BOP dataset, used to load specific weights and object models
       dataset_name: $(var dataset_name)
+      # What type of NN model to use, type of training data used,  'pbr'|'synth+real'
```
Comment (Kotochleb) [inline happypose_examples/config/cosypose_params.yaml]:
````Suggestion
      # Type of neural network model to use. Available 'pbr'|'synth+real' 
```

File: happypose_examples/launch/single_view_demo.launch.py [hunk 5]
```diff
@@ -72,7 +73,12 @@ def generate_launch_description():
         DeclareLaunchArgument(
             "dataset_name",
             default_value="ycbv",
-            description="Which dataset to use for inference.",
+            description="Name of BOP dataset, used to load specific weights and object models.",
+        ),
+        DeclareLaunchArgument(
+            "model_type",
+            default_value="pbr",
+            description="What type of NN model to use, type of training data used, 'pbr'|'synth+real'.",
```
Comment (Kotochleb) [inline happypose_examples/launch/single_view_demo.launch.py:81]:
```suggestion
            description="Type of neural network model to use. Available: 'pbr'|'synth+real'.",
```

File: happypose_examples/launch/multi_view_demo.launch.py [hunk 6]
```diff
@@ -97,7 +98,12 @@ def generate_launch_description():
         DeclareLaunchArgument(
             "dataset_name",
             default_value="ycbv",
-            description="Which dataset to use for inference.",
+            description="Name of BOP dataset, used to load specific weights and object models.",
+        ),
+        DeclareLaunchArgument(
+            "model_type",
+            default_value="pbr",
+            description="What type of NN model to use, type of training data used, 'pbr'|'synth+real'.",
```
Comment (Kotochleb) [inline happypose_examples/launch/multi_view_demo.launch.py:106]:
```suggestion
            description="Type of neural network model to use. Available: 'pbr'|'synth+real'.",
```

File: happypose_examples/launch/common.launch.py [hunk 7]
```diff
@@ -68,7 +68,12 @@ def generate_launch_description():
         DeclareLaunchArgument(
             "dataset_name",
             default_value="ycbv",
-            description="Which dataset to use for inference.",
+            description="Name of BOP dataset, used to load specific weights and object models.",
+        ),
+        DeclareLaunchArgument(
+            "model_type",
+            default_value="pbr",
+            description="What type of NN model to use, type of training data used, 'pbr'|'synth+real'.",
```
Comment (Kotochleb) [inline happypose_examples/launch/common.launch.py:76]:
```suggestion
            description="Type of neural network model to use. Available: 'pbr'|'synth+real'.",
```

File: happypose_examples/config/cosypose_params_multiview.yaml [hunk 8]
```diff
@@ -15,16 +15,16 @@
...
```
Comment (Kotochleb) [inline happypose_examples/config/cosypose_params_multiview.yaml:20]:
```suggestion
      # Type of neural network model to use. Available: 'pbr'|'synth+real'
```

File: happypose_examples/config/cosypose_params.yaml [hunk 9]
```diff
@@ -9,16 +9,16 @@
...
```
Comment (Kotochleb) [inline happypose_examples/config/cosypose_params.yaml:14]:
```suggestion
      # Type of neural network model to use. Available: 'pbr'|'synth+real'
```

**Review by Kotochleb — CHANGES_REQUESTED** [review]

File: happypose_ros/test/test_initialization_params.py (hunk 1 above)
Comment (MedericFourmy) [reply happypose_ros/test/test_initialization_params.py:109]: ok, will remove this line then

**Review by Kotochleb — APPROVED** [review]

**Pull request merged**